File: BackEnd_Learn/Java/todolist/src/main/java/com/leeseojune/todolist/controller/TodoController.java
package com.leeseojune.todolist.controller;

import com.leeseojune.todolist.dto.TodoDTO;
import com.leeseojune.todolist.entity.Todo;
import com.leeseojune.todolist.service.TodoService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import javax.websocket.server.PathParam;

@RestController
@RequiredArgsConstructor
public class TodoController {

    private final TodoService todoService;

    @PostMapping("/post")
    public void addTodo(@RequestBody TodoDTO.Add request){
        todoService.addTodo(request);
    }

    @DeleteMapping("/remove")
    public void delete(@PathParam("id") int id) {
        todoService.delete(id);
    }

    @PatchMapping("/check")
    public void check(@PathParam("id") int id) {
        todoService.check(id);
    }

    @GetMapping("/todo")
    public Iterable<Todo> getAll() {
        return todoService.getAll();
    }

    @PutMapping("/modify/{todoId}")
    public void check(@RequestBody TodoDTO.Modify request, @PathVariable("todoId") int todoId){
        todoService.modify(request, todoId);
    }

}

File: BackEnd_Learn/NestJS/todolist/src/todo/todo.controller.ts
import { Body, Controller, Get, Param, Patch, Post, Put, Query } from '@nestjs/common';
import { todoDTO } from './dto/todo.dto';
import { TodoService } from './todo.service';

@Controller('todo')
export class TodoController {
    constructor(private readonly todoService: TodoService){}

    @Post()
    async post(@Body() body: todoDTO){
        this.todoService.add(body);
    }

    @Get()
    async getAll() {
        return this.todoService.getAll();
    }

    @Patch('check')
    async check(@Query('id') id: number){
        console.log(id);
        this.todoService.check(id);
    }

    @Put('content')
    async change(@Query('id') id: number, @Body() body: todoDTO){
        this.todoService.change(id, body.content);
    }

}
